torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -394,6 +394,31 @@ Node* createSingletonSubgraphAndUpdateAliasing(
       });
 }
 
+std::string truncateStrWithHash(const std::string& s, size_t maxlen) {
+  if (s.size() <= maxlen) {
+    return s;
+  }
+  std::stringstream truncated;
+  truncated << s.substr(0, maxlen);
```

Comment:
I don't know whether to care too much since this is just for logging, but it strikes me as odd that the string returned can actually be maxlen+len(hash) characters long.  I'd rather guarantee getting back something that fits in maxlen.